dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/DataPointWithUnit.kt

```diff
@@ -0,0 +1,21 @@
+package org.dataland.datalandbackend.model
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.dataland.datalandbackend.model.enums.data.QualityOptions
+
+/**
+ * --- API model ---
+ * Fields of a generic data point with unit and its source
+ */
+data class DataPointWithUnit<T>(
+    val value: T? = null,
+
+    @field:JsonProperty(required = true)
+    val quality: QualityOptions,
+
+    val dataSource: CompanyReportReference? = null,
+
+    val comment: String? = null,
+
+    val unit: String?,
+)
```

Comment:
This new data class shares some fields with other "DataPoint"-classes (`DataPoint.kt` and `DataPointAbsoluteAndPercentage.kt`.
Therefore I suggest creating a base class `DataPointBase`, which has all the "base" fields, and then make `DataPointAbsoluteAndPercentage.kt` and `DataPointWIthUnit` inherit from it.

You can look at a similar case for orientation: LKSG and SME both have their own ProductionSite model. But because both share some fields, they both inherit from `ProductionSiteBase.kt`, which is in model/generics.

Reply:
Changed

Reply:
👍